File: python/Kode program/FurnitureItem.py
from HouseOldItem import HouseOldItem
from BrandInfo import BrandInfo
from MaterialInfo import MaterialInfo

class FurnitureItem(HouseOldItem, BrandInfo):
    def __init__(self):
        HouseOldItem.__init__(self)
        BrandInfo.__init__(self)
        self.material = ""
        self.durability = ""
        self.materialInfo = MaterialInfo()

    def setMaterial(self, m):
        self.material = m

    def getMaterial(self):
        return self.material

    def setDurability(self, d):
        self.durability = d

    def getDurability(self):
        return self.durability

    def setMaterialInfo(self, mi):
        self.materialInfo = mi

    def getMaterialInfo(self):
        return self.materialInfo

    def printFurniture(self):
        print("=== Furniture Item ===")
        self.printItem()
        self.printBrand()
        print(f"Material: {self.material}")
        print(f"Durability: {self.durability}")

File: java/Kode program/FurnitureItem.java
public class FurnitureItem extends HouseOldItem {
    private String material;
    private String durability;
    private MaterialInfo materialInfo = new MaterialInfo();
    private BrandInfo brandInfo = new BrandInfo();

    public void setMaterial(String m) { material = m; }
    public String getMaterial() { return material; }

    public void setDurability(String d) { durability = d; }
    public String getDurability() { return durability; }

    public void setMaterialInfo(MaterialInfo mi) { materialInfo = mi; }
    public MaterialInfo getMaterialInfo() { return materialInfo; }

    public void setBrandInfo(BrandInfo bi) { brandInfo = bi; }
    public BrandInfo getBrandInfo() { return brandInfo; }

    public void printFurniture() {
        System.out.println("=== Furniture Item ===");
        printItem();
        brandInfo.printBrand();
        System.out.println("Material: " + material);
        System.out.println("Durability: " + durability);
    }
}
